bin/schema_generator.sh

```diff
@@ -0,0 +1,156 @@
+    git checkout $MPS_BRANCH_SOURCE
+
+    git branch -D $MPS_BRANCH_WORKING
+    git checkout -b $MPS_BRANCH_WORKING
+}
+
+function prepare_metadata() {
+    # Assumes that all schemas under the current directory are valid JSON schema.
+
+    local telemetry_metadata="metadata/telemetry-ingestion/telemetry-ingestion.1.schema.json"
+    local structured_metadata="metadata/structured-ingestion/structured-ingestion.1.schema.json"
+
+    find ./telemetry -type f -exec metadata_merge $telemetry_metadata {} ";"
+    find . -path ./telemetry -prune -o -type f -exec metadata_merge $structured_metadata {} ";"
+}
+
+function filter_schemas() {
+    # Remove metadata
+    rm -rf metadata
+
+    # Pioneer-study is not nested, remove it
+    rm -rf pioneer-study
+
+    # Replace newlines with backticks (hard to do with sed): cat | tr
+    # Remove the last backtick; it's the file-ending newline: rev | cut | rev
+    # Replace backticks with "\|" (can't do that with tr): sed
+    # Find directories that don't match any of the regex expressions: find
+    # Remove them: rm
+    < /app/mozilla-schema-generator/bin/allowlist tr '\n' '`' | \
+        rev | cut -c 2- | rev | \
+        sed -e 's/`/\\\\|/g' | \
+        xargs -I % find . -type f -regextype sed -not -regex '.*/\(%\|metadata/\)/.*' | grep ".bq" | \
+        xargs rm -rf
+}
+
+function commit_schemas() {
+    # This method will keep a changelog of releases. If we delete and newly
+    # checkout branches everytime, that will contain a changelog of changes.
+    # Assumes the current directory is the root of the repository
+
+    find . -name "*.bq" -type f -exec git add {} +
+    git checkout ./*.schema.json
+    git commit -a -m "Interim Commit"
+
+    git checkout $MPS_BRANCH_PUBLISH || git checkout -b $MPS_BRANCH_PUBLISH
+
+    # Keep only the schemas dir
+    find .  -mindepth 1 -maxdepth 1 -not -name .git -exec rm -rf {} +
+    git checkout $MPS_BRANCH_WORKING -- schemas
+    git commit -a -m "Auto-push from schema generation"
+}
+
+function main() {
+    cd $BASE_DIR || exit
+
+    # -1. Setup ssh key and git config
+    setup_git_ssh
+
+    # 0. Install dependencies
+    setup_dependencies
+
+    # 1. Pull in all schemas from MPS
+    clone_and_configure_mps
+    # CWD: /app/mozilla-pipeline-schemas
+
+    # 2. Remove all non-json schemas (e.g. parquet)
+    find . -not -name "*.schema.json" -type f -exec rm {} +
+
+    # 3. Generate new schemas
+    mozilla-schema-generator generate-glean-ping --out-dir . --pretty
+
+    # 4. Add metadata to all json schemas, drop metadata schemas
+    prepare_metadata
+
+    # 5. Add transpiled BQ schemas
+    find . -type f -name "*.schema.json"|while read -r fname; do
+        bq_out=${fname/schema.json/bq}
+        jsonschema-transpiler --type bigquery "$fname" > "$bq_out"
+    done
+
+    # 5b. Keep only allowed schemas
+    filter_schemas
+
+    # 6. Push to branch of MPS
+    cd ../ || exit
+    commit_schemas
+    git push --force
```

Comment:
From a technical perspective, force is not required for our use case (as we saw when accidentally committing to dev). I would recommend removing the force. In the rare case where somebody pushes the generated-schemas branch by hand in the middle of the airflow job's execution we would probably want the automated schema push to fail anyway. Maybe we have branch protection or something enabled now that requires this.